microsoft/PowerToys
# [FancyZones] Split zones-settings: devices -> applied layouts

## Summary of the Pull Request

**What is this about:**

* Keep applied layouts in a separate file.
* Create applied-layouts.json from zones-settings on FZ start
* Save applied layouts in the `custom-layout.json` in Editor.
* Unit tests
* Clean up obsolete code
* Updated parsing/serialization in the editor
* Changed retry loop in the editor

The whole fix helps with #15529 also since the problem there relies on reading from `zones-settings.json`.

**What is included in the PR:** 

**How does someone test / validate:** 

* Verify that `AppData\Local\Microsoft\PowerToys\FancyZones\zones-settings.json` was replaced with `applied-layouts.json`, `custom-layouts.json`, `layout-hotkeys.json` and `layout-templates.json`. 
* Check FZ in general for regressions.

## Quality Checklist

- [x] **Linked issue:** #14782, #15529

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
